### PR TITLE
fix: locationId should strip PUBLIC_PATH prefix

### PR DESCRIPTION
**Pull request by arbrandes**
If `PUBLIC_PATH` were set to anything other than the default '/', and the MFE were actually hosted under that path as intended, `locationId` would be initialized to the wrong value.

### CODE REVIEW

**Comment by arbrandes** [comment]
@leangseu-edx, we need this fix for the Olive release, as Tutor deploys MFEs under a `PUBLIC_PATH` that's different than "/".  Mind taking a peek?  Thanks!

**Comment by nsprenkle** [comment]
@arbrandes, thanks for opening this PR! Can you provide context on how to test?

**Comment by arbrandes** [comment]
@nsprenkle, aside from testing that nothing breaks with the non-path usage of the MFE (i.e., just having it be accessible at `/`, which is the default), the easiest way is via Tutor.  In particular, by testing the [corresponding PR](https://github.com/overhangio/tutor-mfe/pull/89) in tutor-mfe with the code in here.

You can leave the latter to me, though.  What would be more concerning is for the default usage to break.  Testing that is pretty straightforward.  Anything that uses `locationId` would break, and that is pretty much everything, starting with the app's first API call to list ORAs to grade.

So, basically: create a staff graded assignment, submit a response to it as a learner, then as an instructor go to the MFE to see it.  If this works without API errors, then the PR works.

**Comment by arbrandes** [comment]
@openedx/content-aurora, can I get eyes on this one?  Thanks!

File: src/data/constants/app.js [hunk 1]
```diff
@@ -1,4 +1,7 @@
 import { getConfig } from '@edx/frontend-platform';
 
-export const routePath = `${getConfig().PUBLIC_PATH}:courseId`;
-export const locationId = window.location.pathname.slice(1);
+export const getRoutePath = () => `${getConfig().PUBLIC_PATH}:courseId`;
+export const routePath = getRoutePath();
```
Comment (leangseu-edx) [inline src/data/constants/app.js:4]: I don't think this would make a different when declaring a constant. Maybe remove `const routePath` and `const locationId` entirely to prune from errors.

Comment (arbrandes) [reply src/data/constants/app.js:4]: Apologies, I'm not sure I follow: are you observing errors when running this?

Comment (leangseu-edx) [reply src/data/constants/app.js:4]: I am assuming, `getConfig().PUBLIC_PATH` would only valid after `initialize` method finished from `index.jsx` finish running. 
when you assign `export const routePath = getRoutePath();`, it would get run only once during the first run of the code. which probably the time the `initialize()` still unfinished. My suggestion is remove `routePath` and use `get` method for all of them.

**Comment by sarina** [comment]
This was called out in the Core Contributor retro as needing to merge for ~Olive~ Palm!

What are the barriers to merging this? Can I assist in some way?

**Review by dcoa — APPROVED** [review]
This change looks good to me and this change is important for the next release and tutor-mfe compatibility.

**Comment by leangseu-edx** [comment]
Can I get update for my requested change?

**Comment by arbrandes** [comment]
Closing in favor of https://github.com/openedx/frontend-app-ora-grading/pull/167.
